File: articles/trade-inheritance-surveying-and-practical-calculation/go/proportional_share.go
package main

import "fmt"

func main() {
	total := 1200.0
	weights := []float64{2.0, 1.0, 1.0}
	sum := 0.0
	for _, w := range weights {
		sum += w
	}
	for i, w := range weights {
		fmt.Printf("share_%d=%.4f\n", i+1, total*w/sum)
	}
}

File: articles/trade-inheritance-surveying-and-practical-calculation/rust/proportional_share.rs
fn main() {
    let total = 1200.0_f64;
    let weights = [2.0_f64, 1.0, 1.0];
    let sum: f64 = weights.iter().sum();
    for (idx, weight) in weights.iter().enumerate() {
        println!("share_{}={:.4}", idx + 1, total * weight / sum);
    }
}
